File: MATTHEW/leetcode/AIcoding/5.Qwen-3/cpp_solutions/LCP 17-nGK0Fy.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int calculate(string s) {
        int result = 0;
        int num = 0;
        int sign = 1;
        for (char c : s) {
            if (isdigit(c)) {
                num = num * 10 + (c - '0');
            } else if (c == '+') {
                result += sign * num;
                num = 0;
                sign = 1;
            } else if (c == '-') {
                result += sign * num;
                num = 0;
                sign = -1;
            }
        }
        result += sign * num;
        return result;
    }
};

File: leetcode/AIcoding/2.Gemini-2.0/erlang_solutions/LCP 17-nGK0Fy.erl
-module(solution).
-export([calculate/1]).

calculate(S) ->
  calculate(S, 1, 0).

calculate([], A, B) ->
  A * B.

calculate([H|T], A, B) ->
  case H of
    $A -> calculate(T, A * 2, B);
    $B -> calculate(T, A, B + A);
    _ -> calculate(T, A, B)
  end.